diff --git a/mixer/Mixetta.qml b/mixer/Mixetta.qml
index 3c44f90..199713b 100644
--- a/mixer/Mixetta.qml
+++ b/mixer/Mixetta.qml
@@ -1,7 +1,30 @@
 import QtQuick
 import QtQuick3D
+import Qt.labs.animation
+
 Node {
     id: rOOT
+    property real phono1rot: 0
+    property real phono2rot: 0
+    property real microlineRot: 0
+    property real tape1rot: 0
+    property real tape2rot: 0
+
+    property real selectorRot: 0
+    property real microbassRot: 0
+    property real levelRot: 0
+
+    property real leftMeterRot: 0
+    property real rightMeterRot: 0
+
+    property alias phono1sliderValue: rodec_Mixetta_Rodec_Slider_Phono1.value
+
+    property alias tweak1: powerCube.position
+    property real tweak2
+
+//    property alias rot1: rodec_Mixetta_Rodec_Knobs_Aniso_Phono1_wh.rotation
+//    property alias rot2: rodec_Mixetta_Rodec_Knobs_Phono1_wh.rotation
+
     Node {
         id: rootNode__gltf_orientation_matrix_
         rotation: Qt.quaternion(0.707107, -0.707107, 0, 0)
@@ -115,7 +138,7 @@ Node {
                             x: 0.0855515
                             y: -0.0758887
                             z: 0.019
-                            rotation: Qt.quaternion(0.537937, 0, 0, -0.842985)
+                            eulerRotation: Qt.vector3d(0, 0, -115 - rOOT.levelRot)
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Level.mesh"
@@ -165,7 +188,7 @@ Node {
                             x: 0.160732
                             y: -0.0208635
                             z: 0.019
-                            rotation: Qt.quaternion(0.969134, 0, 0, 0.246536)
+                            eulerRotation: Qt.vector3d(0, 0, 28.5 - rOOT.microbassRot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_MicroBass.mesh"
                             materials: [
                                 rodec_Knobs_Aniso_material
@@ -176,7 +199,7 @@ Node {
                             x: -0.081466
                             y: 0.0231433
                             z: 0.0188349
-                            rotation: Qt.quaternion(-0.225125, 0, 0, 0.97433)
+                            eulerRotation: Qt.vector3d(0, 0, -154 - rOOT.microlineRot)
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_MicroLine.mesh"
@@ -189,7 +212,7 @@ Node {
                             x: -0.181457
                             y: 0.0231388
                             z: 0.0188349
-                            rotation: Qt.quaternion(-0.218043, 0, 0, 0.975939)
+                            eulerRotation: rodec_Mixetta_Rodec_Knobs_Phono1.eulerRotation
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Phono1.mesh"
@@ -202,7 +225,7 @@ Node {
                             x: -0.131464
                             y: 0.0232172
                             z: 0.0188349
-                            rotation: Qt.quaternion(0.61635, 0, 0, -0.787472)
+                            eulerRotation: Qt.vector3d(0, 0, -104 - rOOT.phono2rot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Phono2.mesh"
                             materials: [
                                 rodec_Knobs_Aniso_material
@@ -213,6 +236,7 @@ Node {
                             x: 0.160396
                             y: -0.0763736
                             z: 0.019
+                            eulerRotation: rodec_Mixetta_Rodec_Knobs_Power.eulerRotation
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Power.mesh"
                             materials: [
                                 rodec_Knobs_Aniso_material
@@ -223,6 +247,7 @@ Node {
                             x: 0.085511
                             y: -0.0213978
                             z: 0.019
+                            eulerRotation: Qt.vector3d(0, 0, -rOOT.selectorRot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Selector.mesh"
                             materials: [
                                 rodec_Knobs_Aniso_material
@@ -233,7 +258,7 @@ Node {
                             x: -0.0313092
                             y: 0.0231196
                             z: 0.0188349
-                            rotation: Qt.quaternion(0.6984, 0, 0, 0.715707)
+                            eulerRotation: Qt.vector3d(0, 0, 91 - rOOT.tape1Rot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Tape1.mesh"
                             materials: [
                                 rodec_Knobs_Aniso_material
@@ -244,7 +269,7 @@ Node {
                             x: 0.0185902
                             y: 0.0230919
                             z: 0.0188349
-                            rotation: Qt.quaternion(0.132542, 0, 0, 0.991177)
+                            eulerRotation: Qt.vector3d(0, 0, 165 - rOOT.tape2Rot)
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Aniso_Tape2.mesh"
@@ -257,7 +282,7 @@ Node {
                             x: 0.0852899
                             y: -0.076235
                             z: 0.0110243
-                            rotation: Qt.quaternion(0.537937, 0, 0, -0.842985)
+                            eulerRotation: Qt.vector3d(0, 0, -115 - rOOT.levelRot)
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Level.mesh"
@@ -307,7 +332,7 @@ Node {
                             x: 0.160479
                             y: -0.021216
                             z: 0.0110243
-                            rotation: Qt.quaternion(0.969134, 0, 0, 0.246536)
+                            eulerRotation: Qt.vector3d(0, 0, 28.5 - rOOT.microbassRot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_MicroBass.mesh"
                             materials: [
                                 rodec_Knobs_material
@@ -318,7 +343,7 @@ Node {
                             x: -0.0817241
                             y: 0.0227848
                             z: 0.0112006
-                            rotation: Qt.quaternion(-0.225125, 0, 0, 0.97433)
+                            eulerRotation: Qt.vector3d(0, 0, -154 - rOOT.microlineRot)
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_MicroLine.mesh"
@@ -331,20 +356,24 @@ Node {
                             x: -0.181736
                             y: 0.0227965
                             z: 0.0112006
-                            rotation: Qt.quaternion(-0.218043, 0, 0, 0.975939)
+                            eulerRotation: Qt.vector3d(0, 0, -155 - rOOT.phono1rot + rodec_Mixetta_Rodec_Knobs_Phono1_wh.rotation)
                             materials: [
                                 rodec_Knobs_material
                             ]
+                            pickable: true
+                            WheelHandler {
+                                id: rodec_Mixetta_Rodec_Knobs_Phono1_wh
+                            }
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Knobs_Phono2
-                            rotation: Qt.quaternion(0.61635, 0, 0, -0.787472)
+                            eulerRotation: Qt.vector3d(0, 0, -104 - rOOT.phono2rot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Phono2.mesh"
                             materials: [
                                 rodec_Knobs_material
@@ -355,16 +384,36 @@ Node {
                             z: 0.0110243
+                            eulerRotation: Qt.vector3d(0, 0, powerCube.powerOn ? 15 : 0)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Power.mesh"
                             materials: [
                                 rodec_Knobs_material
                             ]
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Knobs_Selector
+                            eulerRotation: Qt.vector3d(0, 0, -rOOT.selectorRot)
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Selector.mesh"
                             materials: [
                                 rodec_Knobs_material
@@ -376,6 +425,7 @@ Node {
                             y: 0.0227769
                             z: 0.0112006
                             rotation: Qt.quaternion(0.6984, 0, 0, 0.715707)
+                            //eulerRotation: Qt.vector3d(0, 0, 91 - rOOT.tape1Rot) // bug: makes it disappear
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Tape1.mesh"
                             materials: [
                                 rodec_Knobs_material
@@ -387,6 +437,7 @@ Node {
                             y: 0.0227602
                             z: 0.0112006
                             rotation: Qt.quaternion(0.132542, 0, 0, 0.991177)
+                            //eulerRotation: Qt.vector3d(0, 0, 165 - rOOT.tape2Rot) // bug: makes it disappear
                             scale.x: 1
                             scale.y: 1
                             source: "meshes/rodec_Mixetta_Rodec_Knobs_Tape2.mesh"
@@ -400,6 +451,7 @@ Node {
                             y: 0.0295795
                             z: -0.00225275
                             source: "meshes/rodec_Mixetta_Rodec_Needle_Left.mesh"
+                            eulerRotation: Qt.vector3d(0, 0, -rOOT.leftMeterRot)
 
                             PrincipledMaterial {
                                 id: rodec_Needle_material
@@ -442,6 +494,7 @@ Node {
                             y: 0.0295795
                             z: -0.00225275
                             source: "meshes/rodec_Mixetta_Rodec_Needle_Right.mesh"
+                            eulerRotation: Qt.vector3d(0, 0, -rOOT.rightMeterRot)
                             materials: [
                                 rodec_Needle_material
                             ]
@@ -455,43 +508,85 @@ Node {
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Slider_MicroLine
-                            y: -0.0575869
+                            y: -0.0575869 - rodec_Mixetta_Rodec_Slider_MicroLine_dh.persistentTranslation.y / 2000
                             source: "meshes/rodec_Mixetta_Rodec_Slider_MicroLine.mesh"
                             materials: [
                                 rodec_Knobs_material
                             ]
+                            pickable: true
+                            DragHandler {
+                                id: rodec_Mixetta_Rodec_Slider_MicroLine_dh
+                            }
+                            BoundaryRule on y {
+                                minimum: -0.055
+                                maximum: 0.01
+                            }
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Slider_Phono1
-                            y: -0.0554197
+                            y: -0.0554197 - rodec_Mixetta_Rodec_Slider_Phono1_dh.persistentTranslation.y / 2000
                             source: "meshes/rodec_Mixetta_Rodec_Slider_Phono1.mesh"
                             materials: [
                                 rodec_Knobs_material
                             ]
+                            pickable: true
+                            DragHandler {
+                                id: rodec_Mixetta_Rodec_Slider_Phono1_dh
+                            }
+                            BoundaryRule on y {
+                                id: slph1br
+                                minimum: -0.055
+                                maximum: 0.01
+                            }
+                            property real value: (y - slph1br.minimum) / (slph1br.maximum - slph1br.minimum)
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Slider_Phono2
-                            y: -0.0647678
+                            y: -0.0647678 - rodec_Mixetta_Rodec_Slider_Phono2_dh.persistentTranslation.y / 2000
                             source: "meshes/rodec_Mixetta_Rodec_Slider_Phono2.mesh"
                             materials: [
                                 rodec_Knobs_material
                             ]
+                            pickable: true
+                            DragHandler {
+                                id: rodec_Mixetta_Rodec_Slider_Phono2_dh
+                            }
+                            BoundaryRule on y {
+                                minimum: -0.065
+                                maximum: 0.001
+                            }
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Slider_Tape1
-                            y: -0.0667343
+                            y: -0.0667343 - rodec_Mixetta_Rodec_Slider_Tape1_dh.persistentTranslation.y / 2000
                             source: "meshes/rodec_Mixetta_Rodec_Slider_Tape1.mesh"
                             materials: [
                                 rodec_Knobs_material
                             ]
+                            pickable: true
+                            DragHandler {
+                                id: rodec_Mixetta_Rodec_Slider_Tape1_dh
+                            }
+                            BoundaryRule on y {
+                                minimum: -0.065
+                                maximum: 0.001
+                            }
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Slider_Tape2
-                            y: -0.066837
+                            y: -0.066837 - rodec_Mixetta_Rodec_Slider_Tape2_dh.persistentTranslation.y / 2000
                             source: "meshes/rodec_Mixetta_Rodec_Slider_Tape2.mesh"
                             materials: [
                                 rodec_Knobs_material
                             ]
+                            pickable: true
+                            DragHandler {
+                                id: rodec_Mixetta_Rodec_Slider_Tape2_dh
+                            }
+                            BoundaryRule on y {
+                                minimum: -0.065
+                                maximum: 0.001
+                            }
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Stereo_Switch
@@ -499,6 +594,9 @@ Node {
                             materials: [
                                 rodec_Knobs_Aniso_material
                             ]
+                            TapHandler {
+                                onTapped: console.log("tapped stereo switch")
+                            }
                         }
                         Model {
                             id: rodec_Mixetta_Rodec_Trans_0
@@ -543,6 +641,11 @@ Node {
                             materials: [
                                 rodec_Trans_material
                             ]
+
+                            PointLight { position: "0.068,0.035,0.010"; brightness: 3; castsShadow: true; color: "lemonchiffon" }
+                            PointLight { position: "0.092,0.035,0.010"; brightness: 3; castsShadow: true; color: "lemonchiffon" }
+                            PointLight { position: "0.145,0.035,0.010"; brightness: 3; castsShadow: true; color: "lemonchiffon" }
+                            PointLight { position: "0.173,0.035,0.010"; brightness: 3; castsShadow: true; color: "lemonchiffon" }
                         }
                     }
                 }
